accession-commons-jpa/src/test/java/uk/ac/ebi/ampt2d/commons/accession/AccessioningServiceStepDefinitions.java

```diff
@@ -0,0 +1,143 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.commons.accession;
+
+import cucumber.api.java.en.And;
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+import org.junit.Ignore;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.ac.ebi.ampt2d.commons.accession.core.exceptions.AccessionDoesNotExistException;
+import uk.ac.ebi.ampt2d.commons.accession.core.exceptions.AccessionMergedException;
+import uk.ac.ebi.ampt2d.commons.accession.core.exceptions.HashAlreadyExistsException;
+import uk.ac.ebi.ampt2d.test.configuration.CucumberTestConfiguration;
+import uk.ac.ebi.ampt2d.test.models.TestModel;
+import uk.ac.ebi.ampt2d.test.testers.AccessioningServiceTester;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.hibernate.validator.internal.util.Contracts.assertTrue;
+
+@Ignore
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = {CucumberTestConfiguration.class})
+@DirtiesContext
+public class AccessioningServiceStepDefinitions {
+
+    @Autowired
+    private AccessioningServiceTester tester;
+
+    @Given("^already accessioned ([\\w-,]+)$")
+    @When("^user submits ([\\w-,]+) to accessioning service$")
+    public void submitObjectsToAccessioningService(String objects) {
+        tester.getOrCreate(toModels(objects));
+    }
+
+    @Then("^user received accessions: ([\\w-,]+)$")
+    public void iShouldReceiveAccessions(String objects) {
+        final String[] accessions = objects.split(",");
+        tester.getSingleVersionResults().assertAccessions(accessions);
+    }
+
+    @When("^user sends patch (\\w+) for accession ([\\w-]+)$")
+    public void userSendsPatchInputForAccessionAccession(String patchData, String accession) {
+        tester.patch(accession, TestModel.of(patchData));
+    }
+
+    @Then("^user should receive version list for ([\\w-]+) with (\\d+) version$")
+    public void userShouldReceivePatchForAccessionsWithVersion(String accession, int totalVersions) {
+        tester.getLastMultipleVersionResult().assertAccession(accession).assertTotalVersions(totalVersions);
+    }
+
+    @And("^patch must have versions increased$")
+    public void patchMustHaveVersionsIncreased() {
+        tester.getLastMultipleVersionResult().assertVersionAreIncreased();
+    }
+
+    @Then("^user receives accession does not exist$")
+    public void userReceivesAccessionDoesNotExistError() {
+        tester.getLastMethodResponse().assertThrow(AccessionDoesNotExistException.class);
+    }
+
+    @When("^user updates ([\\w-]+) patch (\\d+) with ([\\w-]+)$")
```

Comment:
We should homogeneize the vocabulary, either using version or patch but not both (see version in previous methods).